### PR TITLE
migrate test_iosconfig to pytest

### PR DESCRIPTION
**Pull request by frogunder**
### What does this PR do?

migrate test_iosconfig to pytest

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/pytests/unit/modules/test_iosconfig.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        ]
+    )
+    merge_tree = iosconfig.merge_tree(
+        initial_config=running_config, merge_config=merge_config
+    )
+    assert merge_tree == expected_merge_tree
+
+
+def test_merge_text(running_config, merge_config):
+    extected_merge_text = textwrap.dedent(
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_iosconfig.py:197]:
```suggestion
    expected_merge_text = textwrap.dedent(
```

**Review by s0undt3ch — COMMENTED** [review]
Just a typo

File: tests/pytests/unit/modules/test_iosconfig.py (hunk 1 above)
Comment (frogunder) [reply tests/pytests/unit/modules/test_iosconfig.py]: Fixed typo, thanks.

File: tests/pytests/unit/modules/test_iosconfig.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+Test the iosconfig Execution module.
+"""
+
+import textwrap
+
+import pytest
+
+import salt.modules.iosconfig as iosconfig
+
+# Import Salt modules
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_iosconfig.py:11]:
```suggestion
```

File: tests/pytests/unit/modules/test_iosconfig.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+    virtual-service csr_mgmt
+    !
+    ip forward-protocol nd
+    !"""
+    )
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {}
```
Comment (s0undt3ch) [inline tests/pytests/unit/modules/test_iosconfig.py:81]:
```suggestion
    return {isoconfig: {}}
```
?

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: tests/pytests/unit/modules/test_iosconfig.py (hunk 3 above)
Comment (frogunder) [reply tests/pytests/unit/modules/test_iosconfig.py]: Thanks Pedro.

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**